src/seedu/addressbook/Main.java

```diff
@@ -109,9 +109,13 @@ private CommandResult executeCommand(Command command)  {
         try {
             command.setData(addressBook, lastShownList);
             CommandResult result = command.execute();
-            storage.save(addressBook);
+            try {
+                storage.save(addressBook);
```

Comment:
There is no need to nest a try catch block within a try catch block.
 You can have multiple catch blocks for a try block.

``` java
    try {
        command.setData(....)
        ....
        storage.save(addressBook);
    } catch (StorageOperationException  soe) {
        ....
    } catch (Exception e) {
        ....
    }
```

Also recall from tutorial other ways of improving our exception handling =).